flink-runtime/src/main/java/org/apache/flink/runtime/minicluster/MiniCluster.java

```diff
@@ -341,7 +343,9 @@ public void start() throws Exception {
 					RestHandlerConfiguration.fromConfiguration(configuration),
 					resourceManagerGatewayRetriever,
 					blobServer.getTransientBlobService(),
-					commonRpcService.getExecutor(),
+					WebMonitorEndpoint.createExecutorService(
+						configuration.getInteger(RestOptions.SERVER_NUM_THREADS),
+						"DispatcherRestEndpoint"),
```

Comment:
In the `MiniCluster` case, it might be justifiable to use the `commonRpcService.getExecutor`. In particular, if it is configured to use a shared rpc service/`useSingleRpcService == true`. For this work, we would need a `CloseIgnoringExecutorService` which wraps an `Executor` and ignores the shutdown call. This can also be a follow up.

Reply:
Would the following be all that we need?
```
	private static class CloseIgnoringExecutorService extends AbstractExecutorService {

		private final Executor executor;

		public CloseIgnoringExecutorService(Executor executor) {
			this.executor = executor;
		}

		@Override
		public void shutdown() {
		}

		@Override
		public List<Runnable> shutdownNow() {
			return Collections.emptyList();
		}

		@Override
		public boolean isShutdown() {
			return false;
		}

		@Override
		public boolean isTerminated() {
			return false;
		}

		@Override
		public boolean awaitTermination(long timeout, TimeUnit unit) {
			return false;
		}

		@Override
		public void execute(Runnable command) {
			executor.execute(command);
		}
	}
```

Reply:
Is this about making the MIniCluster more light-weight? If so, couldn't we also reduce the thread-pool size to 1 instead?
Re-using another executor means that it will both differ in behavior and thread-debugging (different names) compared to production.

Reply:
Almost. I think we should have a field `terminationFuture`:
```
private static class CloseIgnoringExecutorService extends AbstractExecutorService {

	private final Executor executor;

	private final CompletableFuture<Void> terminationFuture = new CompletableFuture<>();

	public CloseIgnoringExecutorService(Executor executor) {
		this.executor = executor;
	}

	@Override
	public void shutdown() {
		terminationFuture.complete(null);
	}

	@Override
	public List<Runnable> shutdownNow() {
		shutdown();
		return Collections.emptyList();
	}

	@Override
	public boolean isShutdown() {
		return terminationFuture.isDone();
	}

	@Override
	public boolean isTerminated() {
		return terminationFuture.isDone();
	}

	@Override
	public boolean awaitTermination(long timeout, TimeUnit unit) {
		try {
			terminationFuture.get(timeout, unit);
			return true;
		} catch (TimeoutException e) {
			return false;
		}
	}

	@Override
	public void execute(Runnable command) {
		if (terminationFuture.isDone()) {
			throw new RejectedExecutionException();
		} else {
			executor.execute(command);
		}
	}
}
```

Reply:
Yes that is the intention. Decreasing the size of the thread pool would also be a good option.